tests/kernel/common/testcase.yaml

```diff
@@ -6,7 +6,10 @@ tests:
   kernel.common:
     build_on_all: true
   kernel.common.tls:
-    filter: CONFIG_ARCH_HAS_THREAD_LOCAL_STORAGE and CONFIG_TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE
+    # ARCMWDT can't handle THREAD_LOCAL_STORAGE with USERSPACE, see #52570 for details
```

Comment:
Wonder if you can do the same as below (same as TLS test). It would make reverting easier in the future (as in easier to grep).

Reply:
Unfortunately it was done intentionally this way - the `kernel.common.tls` can be build with or without USERSPACE depending on the platform capabilities - so we still want to test it on platform which are not capable to have userspace.

The `kernel.threads.tls.userspace` is always build with USERSPACE (and filtered out on platforms without userspace) - so we can simply use `toolchain_exclude: arcmwdt` for the test case.